### DISCUSSION TITLE
Updating to 4.0, Possibly as a Fork

### DISCUSSION THREAD

**Discussion by ArthurPayne** [GDQuest/learn-gdscript#832] [General]
I recently decided to use _Learn GDScript From Zero_ to get the rust off my programming skills, and learn the changes in Godot 4.0. Since this project is based on 3.5 however, I was thinking I might go through each lesson, look up the changes, and then correct the content in the lessons. Then, I could even contribute to the project with the modifications. I do have those skills, they're just rusty.

I ended up coming to the Github for the project, and checking the GDQuest Discord, but it looks like instead of updating this project, a new remake is being made for 4.0.

I'm a bit behind, and I know 4.0 has been out for a while now. I don't know how the remake project is going, but I'd like to try updating this project in the mean time, since that should go faster than a remake and provide a solution in the interim.

I've yet to actually contribute to an Open Source project, and this seems like something else anyways. I'm thinking I could fork this project and make the changes myself, but that's not contributing, that's just taking it and making my own. Is anyone else doing any updates on this? If someone else already started doing that, I'd like to know. If just updating the lessons and letting someone else do the coding is an option, I could do that too. I'm eager to get back to my own projects after I go through the lessons, but I also like the idea of helping other folks use Godot.

What do you think? Is it okay for me to fork it? Is anyone else doing this? And given the dual license, it doesn't say if the text of the lessons is included in either. I intend to modify them but keep them close to what's already there, will that be permissable?

**Comment by NathanLovato** [comment]
Hi Arthur. Nobody is working on this as far as I know, and it's fine to fork and make changes. The licenses are super permissive, you can change or reuse anything.

The remake depends on new technology as we want to address some limitations of this existing app, and this requires changes to the engine itself. We're getting to the end of R&D now, then come prototypes, a build system, and a couple of things, so there are still some weeks before starting production

**Reply by ArthurPayne** [reply]
I thought that might be the case. If you're pre-production, and this is functional but the information isn't up to date, it'd be easier to fix this in the mean time. But, you're probably busy with other things, right? So I'll see what I can do. Thank you!

**Reply by NathanLovato** [reply]
It's not as simple as updating the information, as to work, this requires a custom build of Godot exposing some parser functions... mainly the one to get errors iirc. It also needs to be fully ported to Godot 4, and every exercise re-written for Godot 4's new GDScript syntax. So it's probably around 2-4 weeks of full-time work if everything goes well.
